src/ccache.c

```diff
@@ -2051,6 +2070,10 @@ from_cache(enum fromcache_call_mode mode, bool put_object_in_manifest)
 		break;
 	}
 
+	if (conf->debug) {
+		debug_end(true);
```

Comment:
Calling `debug_end` explicitly means that we won't get a log e.g. when `failed()` is called. How about conditionally registering `debug_end` as an exit function (see `exitfn_add_nullary`) instead?